### PR TITLE
Fix Password Forms Header Title

### PR DESCRIPTION
**Pull request by mananjadhav**
@Jag96 Can you please review?

### Details
1. Add props to SignIn Layouts to cater to dynamic `welcomeText`
2. Passed `welcomeText` prop to all the forms necessary
3. Updated language files with content for `welcomeText` phrases

### Fixed Issues

$ #4315 

### Tests
1. Checked enter password page for all devices
2. Checked set password page 

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web
enter-password-web

set-password

#### Mobile Web
enter-password-mweb

#### Desktop
enter-password-desktop

#### iOS
enter-password-ios

#### Android
enter-password-android

### CODE REVIEW

**Comment by mananjadhav** [comment]
@Jag96  PR Raised. Can someone verify the Spanish translations? I've picked from Google translate.

File: src/pages/signin/SignInPageLayout/SignInPageLayoutWide.js [hunk 1]
```diff
@@ -15,6 +15,10 @@ const propTypes = {
     /** The children to show inside the layout */
     children: PropTypes.node.isRequired,
 
+    // welcomeText to show the header of the form.
+    // Will change based on the form
```
Comment (jasperhuangg) [inline src/pages/signin/SignInPageLayout/SignInPageLayoutWide.js:19]:
```suggestion
    /** Welcome text to show in the header of the form, changes depending on form type (set password, sign in, etc.) */
```

File: src/pages/signin/SignInPage.js [hunk 2]
```diff
@@ -103,8 +113,8 @@ class SignInPage extends Component {
 SignInPage.propTypes = propTypes;
 SignInPage.defaultProps = defaultProps;
 
-export default withOnyx({
+export default withLocalize(withOnyx({
     account: {key: ONYXKEYS.ACCOUNT},
     credentials: {key: ONYXKEYS.CREDENTIALS},
     session: {key: ONYXKEYS.SESSION},
-})(SignInPage);
+})(SignInPage));
```
Comment (jasperhuangg) [inline src/pages/signin/SignInPage.js:120]: Let's use `compose` for when we wanna combine multiple HOCs together. Something like this:

```js
export default compose(
    withLocalize,
    withOnyx({
        account: {key: ONYXKEYS.ACCOUNT},
        credentials: {key: ONYXKEYS.CREDENTIALS},
        session: {key: ONYXKEYS.SESSION},
    }),
)(SignInPage);
```

File: src/pages/signin/SignInPageLayout/SignInPageLayoutNarrow.js [hunk 3]
```diff
@@ -14,6 +14,10 @@ const propTypes = {
     /** The children to show inside the layout */
     children: PropTypes.node.isRequired,
 
+    // welcomeText to show the header of the form.
+    // Will change based on the form
```
Comment (jasperhuangg) [inline src/pages/signin/SignInPageLayout/SignInPageLayoutNarrow.js:18]:
```suggestion
    /** Welcome text to show in the header of the form, changes depending on form type (set password, sign in, etc.) */
```

File: src/pages/signin/SignInPageLayout/index.js [hunk 4]
```diff
@@ -10,9 +10,9 @@ const propTypes = {
 const SignInPageLayout = props => (
     !props.isSmallScreenWidth
         // eslint-disable-next-line react/jsx-props-no-spreading
```
Comment (jasperhuangg) [inline src/pages/signin/SignInPageLayout/index.js:12]: Doesn't look like there's any props spreading happening here anymore, can you try getting rid of these `eslint-disable-next-line` statements?

File: src/pages/SetPasswordPage.js [hunk 5]
```diff
@@ -82,7 +82,7 @@ class SetPasswordPage extends Component {
     render() {
         return (
             <SafeAreaView style={[styles.signInPage]}>
-                <SignInPageLayout>
+                <SignInPageLayout welcomeText={this.props.translate('welcomeText.phrase4')}>
```
Comment (jasperhuangg) [inline src/pages/SetPasswordPage.js:85]: You'll have to change this when you update the translations as per my other comment.

**Review by jasperhuangg — CHANGES_REQUESTED** [review]
Looks good! Left some minor comments. I've also asked about your translations, will let you know about them when I get a response!

File: src/pages/signin/SignInPageLayout/SignInPageLayoutWide.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (mananjadhav) [reply src/pages/signin/SignInPageLayout/SignInPageLayoutWide.js]: Done

File: src/pages/signin/SignInPage.js [hunk 6]
```diff
@@ -81,10 +84,17 @@ class SignInPage extends Component {
         // - AND an account did not exist or is not validated for that login
         const showResendValidationLinkForm = this.props.credentials.login && !validAccount;
 
+        // Page welcomeText changes based on the type form
+        let welcomeText = this.props.translate('welcomeText.phrase1');
+        if (showPasswordForm) {
+            welcomeText = this.props.translate('welcomeText.phrase5');
+        }
+
```
[the remark this answers is not in the document]
Comment (mananjadhav) [reply src/pages/signin/SignInPage.js]: Done

File: src/languages/en.js [hunk 7]
```diff
@@ -108,6 +108,8 @@ export default {
         phrase1: 'Welcome to the New Expensify! Enter your phone number or email to continue.',
         phrase2: 'Money talks. And now that chat and payments are in one place, it\'s also easy.',
         phrase3: 'Your payments get to you as fast as you can get your point across.',
+        phrase4: 'Welcome back to the New Expensify! Please set your password.',
+        phrase5: 'Welcome back to the New Expensify! Please enter your password.',
```
[the remark this answers is not in the document]
Comment (mananjadhav) [reply src/languages/en.js]: I've moved `phrase4` to `setPasswordPage.passwordFormTitle`.  `phrase5` is retained in `welcomeText` as it is a part of the `SiginInPage` flow. Let me know if you still want me to change it?

File: src/pages/signin/SignInPage.js (hunk 2 above)
Comment (mananjadhav) [reply src/pages/signin/SignInPage.js]: Done. Thanks for the tip.

File: src/pages/SetPasswordPage.js (hunk 5 above)
Comment (mananjadhav) [reply src/pages/SetPasswordPage.js]: Updated translate ref.

File: src/pages/signin/SignInPageLayout/index.js (hunk 4 above)
Comment (mananjadhav) [reply src/pages/signin/SignInPageLayout/index.js]: You're right. No spreading. Removed and no lint error. I am not sure why it was added earlier.

File: src/pages/signin/SignInPageLayout/SignInPageLayoutNarrow.js (hunk 3 above)
Comment (mananjadhav) [reply src/pages/signin/SignInPageLayout/SignInPageLayoutNarrow.js]: Updated

File: src/languages/es.js [hunk 8]
```diff
@@ -108,6 +108,7 @@ export default {
         phrase1: 'Con el Nuevo Expensify, chat y pagos son lo mismo.',
         phrase2: 'El dinero habla. Y ahora que chat y pagos están en un mismo lugar, es también fácil.',
         phrase3: 'Tus pagos llegan tan rápido como tus mensajes.',
+        phrase4: '¡Bienvenido de nuevo al nuevo Expensify! Por favor, introduzca su contraseña.',
```
Comment (jasperhuangg) [inline src/languages/es.js:111]: Heard back about the translations!
```suggestion
        phrase4: '¡Bienvenido de vuelta al Nuevo Expensify! Por favor, elige una contraseña.',
```

File: src/languages/es.js [hunk 9]
```diff
@@ -337,6 +338,7 @@ export default {
         setPassword: 'Configura tu Contraseña',
         passwordsDontMatch: 'Las contraseñas deben coincidir',
         newPasswordPrompt: 'Su contraseña debe tener al menos 8 caracteres, \n1 letra mayúscula, 1 letra minúscula, 1 número.',
+        passwordFormTitle: '¡Bienvenido de nuevo al nuevo Expensify! Por favor establezca su contraseña.',
```
Comment (jasperhuangg) [inline src/languages/es.js:341]:
```suggestion
        passwordFormTitle: '¡Bienvenido de vuelta al Nuevo Expensify! Por favor, elige una contraseña.',
```

**Review by jasperhuangg — DISMISSED** [review]

**Review by jasperhuangg — APPROVED** [review]
Looks good! I heard back about the translations, so I'll commit those and get this merged, thanks!

**Review by Jag96 — APPROVED** [review]

**Pull request merged**